Source/debug.cpp

```diff
@@ -294,6 +295,84 @@ std::string DebugCmdLoadMap(const string_view parameter)
 	return "Welcome to this unique place.";
 }
 
+void WriteLE16(std::ofstream &out, uint16_t val)
+{
+	const uint16_t littleEndian = SDL_SwapLE16(val);
+	char data[2];
+	memcpy(data, &littleEndian, 2);
+	out.write(data, 2);
+}
+
+std::string ExportDun(const string_view parameter)
+{
+	std::ofstream dunFile;
+
+	std::string levelName = fmt::format("{}-{}.dun", currlevel, glSeedTbl[currlevel]);
+
+	dunFile.open(levelName, std::ios::out | std::ios::app | std::ios::binary);
+
+	WriteLE16(dunFile, DMAXX);
+	WriteLE16(dunFile, DMAXY);
+
+	/** Tiles. */
+	for (int y = 0; y < DMAXY; y++) {
+		for (int x = 0; x < DMAXX; x++) {
+			WriteLE16(dunFile, dungeon[x][y]);
+		}
+	}
+
+	/** Padding */
+	for (int y = 16; y < MAXDUNY - 16; y++) {
+		for (int x = 16; x < MAXDUNX - 16; x++) {
+			WriteLE16(dunFile, 0);
+		}
+	}
+
+	/** Monsters */
+	for (int y = 16; y < MAXDUNY - 16; y++) {
+		for (int x = 16; x < MAXDUNX - 16; x++) {
+			uint16_t monsterId = 0;
+			int mid = abs(dMonster[x][y]) - 1;
+			if (mid != -1) {
+				for (int i = 0; i < 157; i++) {
+					if (MonstConvTbl[i] == Monsters[mid].MType->mtype) {
+						monsterId = i + 1;
+						break;
+					}
+				}
+			}
+			WriteLE16(dunFile, monsterId);
+		}
+	}
+
+	/** Objects */
+	for (int y = 16; y < MAXDUNY - 16; y++) {
+		for (int x = 16; x < MAXDUNX - 16; x++) {
+			uint16_t objectId = 0;
+			int oid = abs(dObject[x][y]) - 1;
+			if (oid != -1) {

```

Comment:
Does this work with objects that occupy two tiles (for example Sarcophagus)?
Or get the object added twice?

Reply:
I did notice the issue for one of the Catacomb maps, but didn't sport the double Sarcophagus, but you are right about this as well :)

![image](https://user-images.githubusercontent.com/204594/171695827-109204f7-055b-47d1-aac3-9c422827a5d2.png)

Reply:
Fixed:
![image](https://user-images.githubusercontent.com/204594/171745239-fb3be997-9984-4e7c-99c1-5e3544241ae7.png)